README.md

```diff
@@ -0,0 +1,69 @@
+# Educational-website-and-courses-finder-for-keyword
+Return the filtered useful educational resources based on input keyword and filtering conditions from users
+## Functional Design(secondary version, subject to change)
+### Module: DataCollection
+1. Function: DataCollection
+   - Description: The search process for resources gathering
+   - Input:
+     - keywords: list of keywords for related educational resources finding
+   - Output:
+     - raw_data: unprocessed related resources found through the search, stored in lists for future processing
+   - Functionality: Obtain searching resources from Google Search with keywords
+
+2. Function: Preprocessing(For classifier)
+   - Description: Process the resources for later processing
+   - Input:
+     - raw_data: unprocessed related resources found through the search, stored in lists for future processing
+   - Output:
+     - ripe_data: processed related resources found through the search, format friendly for later modules
+   - Functionality: Turn the rough format of resources into the format and syntax that later modules want
+
+### Module: UsefulnessRanking(This should be a large module, with many incoming supplement functions)
```

Comment:
This module and the below modules (i.e. `UsefulnessRanking` and `FurtherResearch`) can be combined since the only difference are some flags.